IHP/ModelSupport.hs

```diff
@@ -342,14 +342,24 @@ logQuery query parameters = when queryDebuggingEnabled (putStrLn (tshow (query,
 --
 -- Use 'deleteRecords' if you want to delete multiple records.
 deleteRecord :: forall model id. (?modelContext :: ModelContext, Show id, KnownSymbol (GetTableName model), HasField "id" model id, ToField id) => model -> IO ()
-deleteRecord model = do
-    let id = getField @"id" model
+deleteRecord model = get #id model |> delete @model @id
+
+{-# INLINE deleteRecord #-}
+
+-- | Like 'deleteRecord' but using an Id
+--
+-- >>> let project :: Id Project = ...
+-- >>> delete projectId
+-- DELETE FROM projects WHERE id = '..'
+--
+delete :: forall model id. (?modelContext :: ModelContext, Show id, KnownSymbol (GetTableName model), HasField "id" model id, ToField id) => id -> IO ()
```

Comment:
Good addition 👍  Could we name this function `deleteRecordById`? Then it's very clear what's the difference of `deleteRecord` and `deleteRecordById` just by looking at the function name.